src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -1800,27 +1800,15 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
                 }
             }
 
-            GenTree* loadIntrinsic = gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, simdBaseJitType, simdSize);
+            GenTreeHWIntrinsic* loadIntrinsic =
+                gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, simdBaseJitType, simdSize);
             // This operation contains an implicit indirection
             //   it could point into the global heap or
             //   it could throw a null reference exception.
             //
             loadIntrinsic->gtFlags |= (GTF_GLOB_REF | GTF_EXCEPT);
```

Comment:
I think you can delete this line, it should not be necessary now (but check we still set the flags in the constructor for the HWI node).

Reply:
This looks somehow out of scope. I abstracted the common code path from arm64, but didn't meant to modify it. I don't have the environment to verify this change either.

Reply:
`SetMultiReg` and `lvIsMultiRegRet` in fact doesn't do the same thing and replacing would cause failures.

Reply:
`SetMultiReg()` goes on `GenTree` node while `lvIsMultiRegRet` goes on `LclVarDsc`. You missed setting the flag on `LclVarDsc`.